ubiconfig/_impl/loaders/gitlab.py

```diff
@@ -36,6 +36,25 @@ def load(self, file_name):
 
         return UbiConfig.load_from_dict(config_dict, file_name)
 
+    def load_from_cs_label(self, cs_label):
+        # strip any suffixes from label
+        clean_label = cs_label
+        for suffix in ("-rpms", "-source-rpms", "-debug-rpms"):
+            if cs_label.endswith(suffix):
+                clean_label = cs_label.replace(suffix, "")
+
+        # since file could be *.yaml or *.yml, try to load and return
+        # each, capturing any failures
+        load_errs = []
+        for extension in (".yaml", ".yml"):
+            try:
+                return self.load(clean_label + extension)
+            except KeyError as e:
+                load_errs.append(e)
+
+        LOG.error("No configuration file found for label %s at %s", clean_label, self._url)
+        raise load_errs[-1]
```

Comment:
I don't like this approach because you are going to raise a KeyError. You don't need to store any errors actually, will save memory.

If you reach this line is because no file was available, just raise some error like `NoContentSetLabalAvailable`. It's self explanatory.

```py
        for extension in (".yaml", ".yml"):
            try:
                return self.load(clean_label + extension)
            except KeyError as e:
                pass

        LOG.error("No configuration file found for label %s at %s", clean_label, self._url)
        raise NoContentSetLabalAvailable(clean_label)
```